CMakeLists.txt

```diff
@@ -781,6 +781,10 @@ if(developer-options)
 		ENABLED TORRENT_PROFILE_CALLS=1)
 endif()
 
+if (${CMAKE_CXX_STANDARD} STREQUAL "11")
```

Comment:
`CMAKE_CXX_STANDARD` does not contain the C++ version value that the compiler will use in all cases. It is only defined if the user defines it in the command line (or if its value is forced in the CML), and its value is used to initialize the `CXX_STANDARD` property of targets (i.e., if `CMAKE_CXX_STANDARD` is defined when a target is created, its `CXX_STANDARD` property is initialized to that value).

I assume what you want is "if building in C++11 mode, use this flag".

I'm lacking in imagination right now to figure out how to do this automatically, but you can always expose it as a feature:

```cmake
feature_option(cxx11_abi "ensure link compatibility with C++11" OFF)

# ....

if (cxx11_abi)
    target_compile_definitions(torrent-rasterbar PUBLIC TORRENT_CXX11_ABI)
endif()
```

Reply:
ping @zeule ideas?

Reply:
I don't think making it an option makes sense, since the only thing it would affect would be the `.pc` file. @zeule is there a (better) way to detect whether the compiler is in C++11 mode, and if so, add this define to the `.pc` file?

Reply:
@arvidn 

This is not just important for the `.pc` file, but also for the CMake config files. The `TORRENT_CXX11_ABI` must be propagated up to enable the correct usage of the `entry.hpp` header when using both C++11 and C++14 or later.

Reply:
even if this check doesn't work, the patch still opens up the opportunity for the client to fix issues by defining this macro

Reply:
@arvidn That's true. I would suggest dropping the change to the CML, and just merge the rest. Or at least adding a comment like `# TODO: won't work properly unless user explicitly passes "-DCMAKE_CXX_STANDARD" to the configure command line`

Reply:
@arvidn

by the way, you must also enclose the variable in quotes:

`if ("${CMAKE_CXX_STANDARD}" STREQUAL "11")`

 or it will fail when the variable is undefined.